doc/devices.rst

```diff
@@ -104,25 +104,142 @@ Each problem is unique, so it can often be best to choose the default behaviour
-the available GPU memory and the memory requirements of the local state vector, to avoid memory issues and ensure optimal performance.
-By default (`mpi_buf_size=0`), the GPU memory allocated for MPI operations will be the same of size of the local state vector, with a limit of 64 MB(megabytes).
-
-The workflow for the new feature is:
+The ``lightning.gpu`` device allows users to leverage the computational power of multi-node and multi-GPUs for running large-scale simulations. 
+Provided that NVIDIA cuQuantum libraries, a ``CUDA-aware MPI`` library and ``mpi4py`` are properly installed and the path to the ``libmpi.so`` is 
+added to the ``LD_LIBRARY_PATH`` environment variable, the following requirements should be met to enable multi-node and multi-GPU simulations:
+
+1. Both total number of MPI processes and MPI processes per node must be the same and a power of 2. For example, 2, 4, 8, 16, etc.. 
+2. Each MPI process is responsible for managing one GPU. 
+3. The ``mpi`` keyword argument should be set as ``True`` when initializing a ``lightning.gpu`` device.
+
+The workflow for the multi-node/GPUs feature is as follows:
+
+.. code-block:: python
+
+    from mpi4py import MPI
+    import pennylane as qml
+    dev = qml.device('lightning.gpu', wires=8, mpi=True)
+    @qml.qnode(dev)
+    def circuit_mpi():
+      qml.PauliX(wires=[0])
+      return qml.state()
+    local_state_vector = circuit_mpi()
+
+Currently, a ``lightning.gpu`` device with the MPI multi-GPU backend supports all the ``gate operations`` and ``observables`` that a ``lightning.gpu`` device with a single GPU/node backend supports.
+
+By default, each MPI process will return the overall simulation results, except for the ``qml.state()`` and ``qml.prob()`` methods. Each MPI process will only return the local simulation
+results for the ``qml.state()`` and ``qml.prob()`` methods to avoid buffer overflow. It will be the users' responsibity to ensure the safety of data collection for those two methods. Here are examples of collecting
+the local simulation results for ``qml.state()`` and ``qml.prob()`` methods:
+
+The workflow for collecting local state vector (using the ``qml.state()`` method) to ``rank 0`` is as follows:
+
 .. code-block:: python
 
     from mpi4py import MPI
     import pennylane as qml
-    dev = qml.device('lightning.gpu', wires=8, mpi=True, mpi_buf_size=1)
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank() 
+    dev = qml.device('lightning.gpu', wires=8, mpi=True)
     @qml.qnode(dev)
     def circuit_mpi():
       qml.PauliX(wires=[0])
       return qml.state()
     local_state_vector = circuit_mpi()
+    #rank 0 will collect the local state vector
+    state_vector = comm.gather(local_state_vector, root=0)
```

Comment:
Can we use 4 spaces for indenting the Python code here?

Reply:
Done!